src/ngx-smart-modal/tests/components/ngx-smart-modal.component.spec.ts

```diff
@@ -243,4 +243,159 @@ describe('NgxSmartModalComponent', () => {
+    expect((component as any).nsmDialog.last.nativeElement.style.left).toBe('60px');
+    expect(result).toBeTruthy();
+  }));
+
+  it('should not startDrag if modal not draggable', async(() => {
+
+    component.draggable = false;
+
+    //spyOn((component), 'setPosition');
+
+    let result = component.startDrag({} as MouseEvent);
+    expect((component as any).dragging).toBeFalsy();
+    expect(result).toBe(false);
+  }));
+
+  it('should not startDrag if event there is no nsmContent', async(() => {
+    component.draggable = true;
+    (component as any).nsmContent = { length: 0 };
+
+    //spyOn((component), 'setPosition');
+
+    let result = component.startDrag({} as MouseEvent);
+    expect((component as any).dragging).toBeFalsy();
+    expect(result).toBe(false);
+  }));
+
+  it('should not startDrag if event srcElement is null', async(() => {
+    component.draggable = true;
+    (component as any).nsmContent = { length: 1 };
+    const fakeEvent = { srcElement: null };
+
+    //spyOn((component), 'setPosition');
+
+    let result = component.startDrag(fakeEvent as MouseEvent);
+    expect((component as any).dragging).toBeFalsy();
+    expect(result).toBe(false);
+  }));
+
+  it('should not startDrag if clicked element isn\'t .draggable', async(() => {
+    component.draggable = true;
+    (component as any).nsmContent = { length: 1 };
+    const fakeSrcElement = document.createElement('div') as any;
+    const fakeEvent = { srcElement: fakeSrcElement };
+
+    //spyOn((component), 'setPosition');
+
+    const result = component.startDrag(fakeEvent as MouseEvent);
+    expect((component as any).dragging).toBeFalsy();
+    expect(result).toBeFalsy();
+  }));
+
+  it('should not startDrag if content doesn\'t contain clicked element', async(() => {
+    component.draggable = true;
+    const fakeSrcElement1 = document.createElement('div') as any;
+    fakeSrcElement1.classList.add('draggable');
+    const fakeSrcElement2 = document.createElement('div') as any;
+    (component as any).nsmContent = { 'length': 1, 'last': { 'nativeElement': { 'contains': (el: any) => el == fakeSrcElement1 } } };
+    const fakeEvent = { clientX: 10, clientY: 30, srcElement: fakeSrcElement2 };
+
+    //spyOn((component), 'setPosition');
+
+    const result = component.startDrag(fakeEvent as MouseEvent);
+    expect((component as any).dragging).toBeFalsy();
+    expect(result).toBeFalsy();
+  }));
+
+  it('should startDrag', async(() => {
+    component.draggable = true;
+    const fakeSrcElement = document.createElement('div') as any;
+    (component as any).nsmContent = { 'length': 1, 'last': { 'nativeElement': { 'contains': (el: any) => el == fakeSrcElement } } };
+    fakeSrcElement.classList.add('draggable');
+    const fakeEvent = { clientX: 10, clientY: 30, srcElement: fakeSrcElement, preventDefault: () => { } };
+
+    (component as any).setPosition = (positionX: any, positionY: any) => {
+      //nothing
+    };
```

Comment:
why it's here ? I don't understand what you want to do

Reply:
I wanted to skip inner function testing, because I have separate test for it.
I will change this.